tests/v2/validation/nodescaling/scaling_node_driver_rke2_test.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RKE2NodeScalingTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	clustersConfig     *provisioninginput.Config
+}
+
+func (s *RKE2NodeScalingTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *RKE2NodeScalingTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	s.clustersConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, s.clustersConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(s.T(), err)
+
+	newUser.Password = user.Password
+}
+
+func (s *RKE2NodeScalingTestSuite) TestScalingRKE2NodePools() {
+	nodeRoles0 := []machinepools.NodeRoles{
+		{
+			ControlPlane: true,
+			Etcd:         false,
+			Worker:       false,
+			Quantity:     1,
+		},
+	}
+
+	nodeRoles1 := []machinepools.NodeRoles{
+		{
+			ControlPlane: false,
+			Etcd:         true,
+			Worker:       false,
+			Quantity:     1,
+		},
+	}
+
+	nodeRoles2 := []machinepools.NodeRoles{
+		{
+			ControlPlane: false,
+			Etcd:         false,
+			Worker:       true,
+			Quantity:     1,
+		},
+	}
+
+	nodeRoles3 := []machinepools.NodeRoles{
+		{
+			ControlPlane: false,
+			Etcd:         false,
+			Worker:       true,
+			Quantity:     2,
+		},
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles []machinepools.NodeRoles
+		client    *rancher.Client
+	}{
+		{"Scaling control plane machine pool by 1", nodeRoles0, s.client},
+		{"Scaling etcd node machine pool by 1", nodeRoles1, s.client},
+		{"Scaling worker node machine pool by 1", nodeRoles2, s.client},
+		{"Scaling worker node machine pool by 2", nodeRoles3, s.client},
+	}
+
+	for _, tt := range tests {
+		s.clustersConfig.NodesAndRoles = tt.nodeRoles
+
+		clusterID, err := clusters.GetV1ProvisioningClusterByName(s.client, s.client.RancherConfig.ClusterName)
+		require.NoError(s.T(), err)
+
+		s.Run(tt.name, func() {
+			ScalingRKE2K3SNodePools(s.T(), s.client, clusterID, s.clustersConfig)
+		})
+	}
+}
+
+func (s *RKE2NodeScalingTestSuite) TestScalingRKE2NodePoolsDynamicInput() {
+	clusterID, err := clusters.GetV1ProvisioningClusterByName(s.client, s.client.RancherConfig.ClusterName)
```

Comment:
can you put logic here like the provisioning tests that will skip if the input is nil,